### ISSUE TITLE
Simulating reads and misalignments

### ISSUE THREAD

**Issue by pesho-ivanov** [ekimb/mapquik#17]
Thank you for the work you put on reproducibility.

Nevertheless, I am puzzled while trying to:

1) reproduce your results -- I am stuck with an issue with pbsim+paftools lh3/minimap2#1121),  I am wondering whether you had issues with NaN's in .maf by paftools (as I described in the minimap2 issue) and whether you considered using a newer version of pbsim (v2 or v3).

2) evaluate mapquik on a simulated HiFi dataset using the [Eskemap](https://github.com/medvedevgroup/eskemap) pipeline -- `mapquik` produces unexpectedly many misalignments.  I simply ran `mapquik` with default parameters on the chm13 Y-chromosome, following the evaluation pipeline of `Eskemap`, and produced only 1306 alignments for 6938 reads. Do you think the produced reads are somehow fundamentally different to those by pbsim and should I change the parameters for `mapquik` for this reason? Here is a distilled version of `Eskemap`'s pipeline:
```
wget 'https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=nucleotide&id=NC_060948.1&rettype=fasta&retmode=text' -O t2thumanChrY.fasta"
./scripts/simReads.py -dp 10 -lmn 100 -lmx 1000000 -lavg 9000 -ls 7000 -r t2thumanChrY.fasta -sr 0.0001 -dr 0.001 -ir 0.0009 -sd 7361077429744071834 -o reads.fasta
mapquik reads.fasta --reference genomes/t2thumanChrY.fasta
```

**Comment by rchikhi** [comment]
hi Pesho

1. I responded to the issue in the minimap2 repo
2. could you try with 24kbp reads please? (and with pbsim..) This is what we mostly evaluated with. If the issue persists, we'll look into it.

thanks for reporting it though!

Rayan

**Comment by pesho-ivanov** [comment]
Thank you, Rayan!

1. Your response helped me resolve the first issue.
2. I tested mapquik (up to date, last commit Oct 7) on several HG references with 10kbp and 24kbp reads with 0.1% and 1% errors (using pbsim) but the accuracy (according to mapeval) keeps being extremely low (>99% wrong alignments). Minimap2 on the same data produces very low wrong alignments (<0.2%). I attach what is needed to reproduce. Will be grateful if you could see and let me know what the problem is.

Raw data:
[chm13-chr1.fa](https://oficina.somnuvol.coop/s/YkbqnFmmGspRx3S) (~200MB)

Generated reads and outputs:
[reads-chm13-chr1-a0.99-d1.fa](https://oficina.somnuvol.coop/s/rde5eLR3ftnMbLE) (~100MB)
[minimap.paf](https://oficina.somnuvol.coop/s/kgR3ogfmrkHpEDS)
[mapquik.paf](https://oficina.somnuvol.coop/s/wJeLAHjnrEXTkfD)

Terminal:
[mapquik.txt](https://github.com/ekimb/mapquik/files/13844381/mapquik.txt)
[minimap2.txt](https://github.com/ekimb/mapquik/files/13844382/minimap2.txt)
[pbsim.txt](https://github.com/ekimb/mapquik/files/13844383/pbsim.txt)

**Comment by rchikhi** [comment]
Hi Pesho, please make sure your reference genome is not a multi-line FASTA (`seqtk seq -AU`) as per the readme

**Comment by pesho-ivanov** [comment]
Thank you, Rayan! This solves the issue.

I got mislead by the [supplementary](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10538364/bin/Supplemental_Material.pdf) of your paper according to which you run all tools on the same reference file while in reality you seem to modify it.

**Comment by rchikhi** [comment]
yes indeed, sorry about that.  I view mapquik as a F1 racecar: very quick prototype but would need many quality of life improvements for real use.
